src/pages/ScoreReport.vue

```diff
@@ -598,6 +604,28 @@ const getScoresAndSupportFromAssessment = ({
   };
 };
 
+const computedProgressData = computed(() => {
+  if (!assignmentData.value) return [];
+  return assignmentData.value.map(({ assignment }) => {
+    const progress = assignment.assessments.reduce((acc, assessment) => {
+      const status = assessment.optional
+        ? 'optional'
+        : assessment.completedOn !== undefined
+        ? 'completed'
+        : assessment.startedOn !== undefined
+        ? 'started'
+        : 'assigned';
+
+      acc[assessment.taskId] = { value: status };
+      return acc;
+    }, {});
+    return {
+      userPid: assignment.userData?.assessmentPid, // Assuming user contains a `username` property
+      progress,
+    };
```

Comment:
Totally optional, but we could consider destructuring the `assignment` object to make the code more readable: 

```suggestion
  return assignmentData.value.map(({ assignment: { assessments, userData } }) => {
    const progress = assessments.reduce((acc, assessment) => {
      const status = assessment.optional
        ? 'optional'
        : assessment.completedOn
        ? 'completed'
        : assessment.startedOn
        ? 'started'
        : 'assigned';

      acc[assessment.taskId] = { value: status };
      return acc;
    }, {});
    return {
      userPid: userData?.assessmentPid
      progress,
    };
```